### PR TITLE
Feat: animation upgrade

### PR DESCRIPTION
**Pull request by jung-han**
### Please check if the PR fulfills these requirements
- [ ] It's submitted to right branch according to our branching model
- [ ] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

* use name for animation 

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/component/boxStackSeries.ts [hunk 1]
```diff
@@ -208,15 +207,17 @@ export default class BoxStackSeries extends BoxSeries {
...
-
-        if (isNumber(barLength)) {
-          seriesModels.push({
-            type: 'rect',
-            color: colors[seriesIndex],
-            value,
-            ...this.getAdjustedRect(seriesPos, dataPosition, barLength, columnWidth),
-          });
-        }
+        const name = seriesRawData[seriesIndex].name;
```
Comment (lja1018) [inline src/component/boxStackSeries.ts:210]: `const { name } = seriesRawData[seriesIndex];`는 어떨까요?

File: src/component/component.ts [hunk 2]
```diff
@@ -173,22 +175,65 @@ export default abstract class Component {
...
+      const modelNames = [...new Set(models.map(({ name }) => name))];
+      const targetNames = [...new Set(targetModels.map(({ name }) => name))];
+      const same = isSameArray(modelNames, targetNames);
+
+      if (!same) {
+        if (currentModels.length > targetModels.length) {
+          return models.filter(({ name }) => includes(targetNames, name));
+        }
+
+        if (currentModels.length < targetModels.length) {
```
Comment (lja1018) [inline src/component/component.ts:189]: 무조건 `currentModels.length < targetModels.length`인 경우라 if문이 필요없지 않을까요?

File: src/component/component.ts [hunk 3]
```diff
@@ -173,22 +175,65 @@ export default abstract class Component {
...
-    } else if (currentModels.length > targetModels.length) {
-      models.splice(targetModels.length, currentModels.length);
+      return [...currentModels, ...targetModels.slice(currentModels.length, targetModels.length)];
+    }
+
+    if (currentModels.length > targetModels.length) {
+      return currentModels.slice(0, targetModels.length);
     }
+
+    return models;
```
Comment (jungeun-cho) [inline src/component/component.ts:225]: if/if사용해서 바로 return하기보다는 할당하여 계산된 값을 마지막에 반환하는게 어때요?
if / else if 사용하구요.

**Review by jungeun-cho — APPROVED** [review]
[8/19] 리뷰완료합니다. 수고하셨어요~ :)
라인 튜플데이터 카테고리 관련해서 범례 동작 확인부탁합니다.

**Pull request merged**